### ISSUE TITLE
Website design 

### ISSUE THREAD

**Issue by SaraVieira** [SaraVieira/webdev.wtf#3]
I'm not a designer 

I say before posting anything in here 😂

**Comment by stereobooster** [comment]
https://wireframe.cc/1Om8tz

<img width="1099" alt="screen shot 2018-12-11 at 14 32 04" src="https://user-images.githubusercontent.com/179534/49804009-c55a2e00-fd51-11e8-9b7b-3f3ff219ec5a.png">

Not a designer either, but i guess we need to agree on elements and pages first, right? So designer would know. We have two pages:

- index - list of all terms. It may have
  - search bar
  - pagination or infinite scroll or list of starting letters
  - list of terms (links to articles)
- article - page dedicated to one term. It may have
  - Title - header
  - Definition - text block, may include images, or interactive "widgets" if we will use MDX and custom components
  - Section with structured data, like tags or also know as...  (optional)
  - Main image (optional) 

WDYT?

**Comment by SaraVieira** [comment]
I was thinking the main page would focus more on the search bar , basically would mostly only have that and a list of the most searched terms 

Then there could be a page with all the terms , what do you think ?

**Comment by stereobooster** [comment]
Yeah, when I wrote index, I meant alphabetic index, like they have in books not a root page.

As of root page (or landing page) I agree - search box in the middle of the page and navigation links (alphabetic index, source code on Github).

About search box. You think about combobox or separate page with search results?

**Comment by stereobooster** [comment]
| ![](https://user-images.githubusercontent.com/179534/49867579-34e02400-fe0b-11e8-8520-2323b3a50752.png) | ![](https://user-images.githubusercontent.com/179534/49867580-3578ba80-fe0b-11e8-8035-b69d4d4369b6.png) | ![](https://user-images.githubusercontent.com/179534/49867581-3578ba80-fe0b-11e8-92c7-f3f47266203d.png) |
|---------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------|
| Landing page: logo, searchbox, links                                                                    | as soon as user focus searchbox, it slides to the top of the page                                       | as soon as user types something in, search results apper under search box                               |

| ![](https://user-images.githubusercontent.com/179534/49868256-298df800-fe0d-11e8-89b2-aa1f044e7206.png) | ![](https://user-images.githubusercontent.com/179534/49868257-298df800-fe0d-11e8-8a2d-0403fa45a1b6.png) |
|---------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------|
| Index page                                                                                              | Definition page                                                                                         |

**Comment by SaraVieira** [comment]
Dude this is perfect 😍

I will try to turn this into an ui :)

**Comment by SaraVieira** [comment]
You are my hero

**Comment by SaraVieira** [comment]
As much as I like dark themes I feel like this website is better suited for a light one right ?

**Comment by jsnhall** [comment]
I would agree that a light theme would be better suited for this.

**Comment by SaraVieira** [comment]
I would then add a dark theme switch that would be on by default if you had the Mac os dark mode , make sense ?

**Comment by SaraVieira** [comment]
Here is my first attempt

![webdev wtf](https://user-images.githubusercontent.com/1051509/49903313-b44b1300-fe5e-11e8-92dc-476d43ee6355.png)


File is open for viewing and editing:

https://www.figma.com/file/Ez7bYjrNe54M23AIdh2Feqrt/Webdev.wtf?node-id=0%3A1

**Comment by stereobooster** [comment]
I like the idea of example searches under searchbox

**Comment by cdvillard** [comment]
It should be purple. Because. 😋

**Comment by SaraVieira** [comment]
That could actually be the accent color that I am missing so far 

I'm in on purple because it's the gay color I have no idea what your reason is but glad we figured this out ahah

**Comment by cdvillard** [comment]
@stereobooster, instead of on focus, could the search be on input? It's a nitpick, but it might be visually jarring to have so much whitespace if nothing is being typed in.

**Comment by stereobooster** [comment]
We need to experiment with real component. We can build theories, but practice can be different

**Comment by cdvillard** [comment]
@SaraViera, maybe the suggestions can be most searched by category (FE, BE, DB, Languages)?

**Comment by stereobooster** [comment]
I would say for start it can be hardcoded values. In the future it indeed can be top searches. (Originally question was for Sara. Welp...)

**Comment by SaraVieira** [comment]
That would require three or four rows under the search and I think that would actually be nice since there is so much white space :)

@stereobooster I will dig into starting on this part of the search since I don't have that much experience with these animations and it needs to be smooth

**Comment by stereobooster** [comment]
If you have some initial boilerplate I also can give it a try.
